src/pages/index.js

```diff
@@ -1,89 +1,92 @@
-const addNewCardButton = document.querySelector(".profile__add-button");
-
-//PREVIEW IMAGE EL
-const previewImageElement = document.querySelector(".modal__image-element");
-const previewImageElementTitle = document.querySelector(
-  ".modal__image-caption"
-);
-//profile inputs
-const profileTitleInput = document.querySelector("#profile-title-input");
-const profileDescriptionInput = document.querySelector(
-  "#profile-description-input"
-);
+//Getting and displaying cards
+
+let section;
+api
+  .getInitialCards()
+  .then((cardData) => {
+    section = new Section(
+      {
+        items: cardData,
+        renderer: (cardData) => {
+          const cardElement = createCard(cardData);
+          section.addItem(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    section.renderItems();
+  })
+  .catch((err) => console.error("Error rendering cards:", err));
 
-//EVENT HANDLERS
 function handleImageClick(data) {
   popupImage.open(data);
 }
-
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.name, inputValues.about);
-  profileEditPopup.close();
-}
-
-// //EVENT LISTENERS
-profileEditButton.addEventListener("click", () => {
-  const data = userInfo.getUserInfo();
-  profileTitleInput.value = data.name;
-  profileDescriptionInput.value = data.about;
-  profileEditPopup.open();
-});
-
 addNewCardButton.addEventListener("click", () => {
   newCardPopup.open();
 });
 
-// functions
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDelete,
+    handleLikeButton
+  );
   const cardElement = card.getView();
   return cardElement;
 }
 
 function handleAddCardFormSubmit(data) {
-  const { name, link } = data;
-  const cardData = { name: name, link: link };
-  const cardElement = createCard(cardData);
-  section.addItem(cardElement);
-  newCardPopup.close();
-  newCardPopup.reset();
-  addCardFormValidator.handleDisableButton();
-}
-
-//instantiations
-const profileEditFormValidator = new FormValidator(config, profileEditForm);
-profileEditFormValidator.enableValidation();
-
-const addCardFormValidator = new FormValidator(config, addCardForm);
-addCardFormValidator.enableValidation();
-
-const section = new Section(
-  {
-    items: initialCards,
-    renderer: (cardData) => {
-      const cardElement = createCard(cardData);
+  newCardPopup.renderLoading(true);
+  api
+    .addCard(data.name, data.link)
+    .then((data) => {
+      const cardElement = createCard(data);
       section.addItem(cardElement);
-    },
-  },
-  ".cards__list"
-);
-
-section.renderItems();
+      newCardPopup.reset();
+      addCardFormValidator.handleDisableButton();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    })
+    .finally(() => {
+      newCardPopup.renderLoading(false);
+      newCardPopup.close();
```

Comment:
* You haven't corrected my remark https://github.com/lydiacodes04/se_project_aroundtheus/pull/1/files#r1639433841  You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) 


**Please, correct it everywhere**